lib/improver/utilities/cube_manipulation.py

```diff
@@ -832,3 +833,36 @@ def enforce_coordinate_ordering(
     elif anchor == "end":
         cube.transpose(remaining_coords+coord_dims)
     return cube
+
+
+def clip_cube_data(cube, result):
+    """Apply np.clip to data in a resulting cube to ensure that the limits do
+    not go beyond the limits of the original input cube.
+
+    Args:
+        cube (iris.cube.Cube):
+            The cube from which data has been taken to be processed.
+        result (iris.cube.Cube):
+            The cube that has been processed and contains data that is to be
+            clipped.
+    Returns:
+        result_slices (iris.cube.Cube):
+            The processed cube with the data clipped to the limits of the
+            original preprocessed cube.
+    """
+    input_max = cube.data.max()
+    input_min = cube.data.min()
+    original_attributes = cube.attributes
+    original_methods = cube.cell_methods
+
+    result_slices = iris.cube.CubeList()
+    for result_slice in result.slices([result.coord(axis='y'),
+                                       result.coord(axis='x')]):
+        result_slice.data = np.clip(result_slice.data, input_min, input_max)
+        result_slices.append(result_slice)
+
+    result_slices = result_slices.merge_cube()
+    result_slices.cell_methods = original_methods
+    result_slices.attributes = original_attributes
+    result_slices = check_cube_coordinates(result, result_slices)
```

Comment:
I think from the for loop to the check_cube_coordinates is a good example of code that should go centrally and then to Iris. Al we want to do is slice a cube and put it back together again - seems really generic!

Reply:
We use this in lots of places, and I could indeed see it being made more general. As we need to get this in I'm going to put something up in a separate PR.